added manual_inspection script to src.

navjot-nangia opened the pull request:
This script:

- goes through the inspection results saved in long term storage for a field batch
- inspects the quality of segmentation masks generated
- saves the results in a csv format to the long term storage batch

The scirpt is heavily inspired by similar inspection from the [semi-field inspection script](https://github.com/precision-sustainable-ag/SemiF-Preprocessing/blob/4432d04abcefe8d9f08a2d34cba9dc573caf8858/src/inspect_images.py).

navjot-nangia committed:
added manual_inspection script to src.

This script allows to go through the inspection results saved in long term storage for field batches and inspects the quality of sgementation masks generated. It saves the results in a csv format to the long term storage batch.

navjot-nangia committed:
added manual_inspection script to src.

This script allows to go through the inspection results saved in long term storage for field batches and inspects the quality of sgementation masks generated. It saves the results in a csv format to the long term storage batch.

navjot-nangia committed:
added delay before displaying flagged images in manual inspection

I noticed that if press 'y' and enter for review flagged images, it crashes.
So, i added a delay of 10 seconds.

navjot-nangia committed:
added delay before displaying flagged images in manual inspection

I noticed that if press 'y' and enter for review flagged images, it crashes.
So, i added a delay of 10 seconds.

navjot-nangia committed:
added manual_inspection in the pipeline. changed the batch_id

navjot-nangia committed:
added manual_inspection in the pipeline. changed the batch_id

mkutu commented on src/manual_inspection.py:
```diff
@@ -0,0 +1,183 @@
+GITHUB_REPO_URL = "https://github.com/precision-sustainable-ag/Field-AnnotationPipeline/issues"
+
+LABEL_OPTIONS = {
+                "1": "Good Mask",
```
Use "pass" instead of good mask

mkutu commented on src/manual_inspection.py:
```diff
@@ -0,0 +1,183 @@
+    def _display_instructions(self):
+        """Prints instructions for user input."""
+        print("\n--- Segmentation Quality Assessment ---")
+        print_labels = {
```
If you're not using emojis, just use the global LABEL_OPTIONS

navjot-nangia replied:
I added emojis.

mkutu commented on src/manual_inspection.py:
```diff
@@ -0,0 +1,183 @@
+                cv2.destroyAllWindows()
+                return self.csv_file  # Save progress and exit
+
+            self.results.append([self.batch_id, img_path.stem, label, self.timestamp, self.user, self.longterm_inspection_dir])
```
The last entry should just be the name of the LTS directory. In this case longterm_images3. So Path(self.cfg.paths.longterm_storage).name

mkutu commented on src/manual_inspection.py:
```diff
@@ -0,0 +1,183 @@
+        print("Mention the flagged images and describe the issues.")
+        if input("Would you like to review the flagged images for screenshots? (y/n): ").strip().lower() == 'y':
+            print("Waiting to allow the image viewer to load...")
+            time.sleep(10)
```
Not sure what this is for. You're delaying the entire script no?

navjot-nangia replied:
I observed that if i press 'y' and enter too quick, the image viewer crashes. That's why i added it. Others might find a similar issue. It is only one time '10' second delay.

mkutu replied:
How are you using x11 forwarding?

In another separate terminal (not vs code) ssh into sunny using

ssh -X -C <username>@SUNNY.ece.ncsu

navjot-nangia replied:
I am using a separate terminal and ssh -X  ___ @SUNNY.ece.ncsu

navjot-nangia committed:
updates label and csv file

- added emojis to labels
- updated last entry in csv file
- changed "Good Mask" to "Pass"

navjot-nangia committed:
updates label and csv file

- added emojis to labels
- updated last entry in csv file
- changed "Good Mask" to "Pass"

mkutu commented:
I'm working on the semif version and I'm realizing we need a back functionality and a final confirmation to save from the user. So at the very end we ask if they want to save or start over. 
So add:
1. Go back functionality
2. Final confirmation to save or start over.

mkutu commented:
For the semif example:
I've added the back functionality [here](https://github.com/precision-sustainable-ag/SemiF-Preprocessing/blob/bc61fd4a59e622dda9340b6b61b49207ba8bb8dd/src/inspect_images.py#L119)

and added the final user confirmation [here](https://github.com/precision-sustainable-ag/SemiF-Preprocessing/blob/bc61fd4a59e622dda9340b6b61b49207ba8bb8dd/src/inspect_images.py#L135)

navjot-nangia committed:
added 'Back' option, improve instructions, and confirm save results: copied from the semif updates

navjot-nangia committed:
added 'Back' option, improve instructions, and confirm save results: copied from the semif updates

mkutu committed:
Merge branch 'manual_inspection' of https://github.com/precision-sustainable-ag/Field-AnnotationPipeline into manual_inspection

mkutu committed:
minor update environment, config default for ease of use

mkutu committed:
simple user sanity check for running pipeline

mkutu committed:
several adjustments and corrections to manual_inspections

mkutu commented:
~~I'll do this~~
~~can you add document in the readme to run the inspection script? You can refer to my documentation in semif. Please be sure not to just copy and paste without making sure the relevant adjustments that we've made are properly represented~~

Make documentation another issue.
